### ISSUE TITLE
[Bug] Wrong IconButton positioning in v3.0.3

### ISSUE THREAD

**Issue by SiboVG** [weisJ/darklaf#347]
**Describe the bug**
When using the latest snapshot of darklaf, v3.0.3, the positioning of some of my buttons has become messed up. I've been able to trace the issue down to commit 45679f69adf5bbb1901425d41806a8de96dc35db.

**To Reproduce**
Issue is happening for the following widgets:
```Java
String[] SCALE_LABELS = ...;
JButton zoomOutButton = new JButton(Icons.ZOOM_OUT);
JComboBox scaleSelectorCombo = new JComboBox<>(SCALE_LABELS);
scaleSelectorCombo.setEditable(true);
JButton zoomInButton = new JButton(Icons.ZOOM_IN);
```

**Screenshots**
<img width="1566" alt="Screenshot 2023-07-16 at 13 13 16" src="https://github.com/weisJ/darklaf/assets/11031519/f18850b8-7a4b-4548-847e-52b6092f9baa">

**Additional Information:**
 - OS: macOS
 - OS Version: 13.4.1
   - For issues with frame decorations on Windows please provide the explicit [Windows build number](https://docs.microsoft.com/en-gb/windows/release-information/).
- Darklaf Version: 3.0.3-SNAPSHOT
  - **Note**: If you are using the `latest.integration` version please try to replicate the issue with the latest stable release.

**Comment by SiboVG** [comment]
UPDATE: I'm able to resolve this issue when I remove the 20 offset from the `shadowSize`

```Java
// This causes an issue
shadowSize = UIManager.getInt("Button.shadowHeight") + 20;

// This works!
shadowSize = UIManager.getInt("Button.shadowHeight");
```

**Comment by weisJ** [comment]
Yes that are some left over debugging remains. I have already fixed it locally but didn’t push the changes yet due to some unfinished other changes.
